modules/core/src/core/transform.js

```diff
@@ -221,4 +304,118 @@ export default class Transform {
+  }
+
+  // scan and update vertex shader for texture atrributes.
+  _processVertexShader(vs) {
+    const uniforms = {};
+    if (!this.hasSourceTextures) {
+      return {vs, uniforms};
+    }
+    let updatedVs = vs;
+    let mainInstructions = null;
+    const texAttributeNames = Object.keys(this.sourceTextures[this.currentIndex]);
+    if (texAttributeNames.length > 0) {
+      const vsLines = updatedVs.split('\n');
+      const updateVsLines = vsLines.slice();
+      const defNameMap = texAttributeNames.reduce((result, name) => {
+        const def = `${name};`;
+        result[def] = name;
+        return result;
+      }, {});
+      const sampleInstructions = [];
+      vsLines.forEach((line, index, lines) => {
+        const updated = this._processTextureAttributes(line, defNameMap);
+        if (updated) {
+          const {updatedLine, sampleInstruction, samplerUniforms} = updated;
+          updateVsLines[index] = updatedLine;
+          sampleInstructions.push(sampleInstruction);
+          Object.assign(uniforms, samplerUniforms);
+        }
+      });
+      updatedVs = updateVsLines.join('\n');
+      mainInstructions = mainInstructions ?
+        mainInstructions + sampleInstructions : sampleInstructions;
+    }
+
+    if (mainInstructions) {
+      updatedVs = updatedVs.replace(REGEX_START_OF_MAIN, match => match + mainInstructions);
+    }
+
+    return {vs: updatedVs, uniforms};
+  }
+
+  // build required definitions, sample instructions and uniforms for each texture attribute
+  _processTextureAttributes(line, defNameMap) {
+    const samplerUniforms = {};
+    const words = line.replace(/^\s+/, '').split(/\s+/);
+    const [qualifier, size, definition] = words;
+    if (
+      (qualifier !== 'attribute' && qualifier !== 'in') ||
+      !size ||
+      !definition
+    ) {
+      return null;
+    }
+    // check if a texture source is specified for this attribute
+    const name = defNameMap[definition];
+    if (name) {
+      assert(this.sourceTextures[0][name]);
+      const updatedLine = `\
+uniform sampler2D transform_uSampler_${definition} \// ${line}
+uniform vec2 transform_uSize_${definition}`;
+      const channels = sizeToChannels(size);
+      const sampler = `transform_uSampler_${name}`;
+      const texSize = `transform_uSize_${name}`;
+      const sampleInstruction =
+        `  ${size} ${name} = transform_getInput(${sampler}, ${texSize}).${channels};\n`;
+
+      const samplerName = `transform_uSampler_${name}`;
+      const sizeName = `transform_uSize_${name}`;
+      const {width, height} = this.sourceTextures[this.currentIndex][name];
+      samplerUniforms[samplerName] = this.sourceTextures[this.currentIndex][name];
+      samplerUniforms[sizeName] = [width, height];
+
+      return {updatedLine, sampleInstruction, samplerUniforms};
+    }
+    return null;
+  }
+}
+
+// Utility methods
+
+function sizeToChannels(size) {
+  switch (size) {
+  case 'float': return 'x';
```

Comment:
can it only be `floats` and `vecs`?

Reply:
I will add other types in future PRs as needed.